### PR TITLE
Command to execute a snippet #35

### PR DESCRIPTION
**Pull request by sambhavnoobcoder**
## Description 
This PR addresses issue #35  by implementing a new 'run' command in the Pieces CLI that allows users to directly execute their saved bash snippets in the terminal.

## Problem:
Users want a quick way to execute their saved bash snippets without having to manually copy and paste them into the terminal.

## Solution:
We've implemented a new ExecuteCommand class that allows users to execute a snippet by its name directly from the Pieces CLI. This is achieved through two new files: run_shell_command.py and run_bash_command.py.

## Changes:

1. run_shell_command.py (new file):
   - Implements ExecuteCommand class for executing shell commands
   - Includes methods for asset retrieval, selection, and execution
   - Uses subprocess to run commands with shell=True

2. run_bash_command.py (new file):
   - Similar to run_shell_command.py, but executes commands in bash
   - Uses subprocess to run commands with ['bash', '-c', command]

## Key features:
- Users can now run snippets directly using the Pieces CLI
- Implements both shell and bash execution of snippet content
- Handles errors and displays execution results
- Maintains existing functionality for browsing and selecting snippets
- Limits the number of assets displayed (default 10, customizable)

## Implementation details:
- Both files use the ExecuteCommand class with similar structure
- The main difference is in the execute_in_shell and execute_in_bash methods
- Utilizes existing Pieces functionality (AssetsCommandsApi, ListCommand, etc.)
- Integrates with PiecesSelectMenu for asset selection

### CODE REVIEW

**Comment by Bishoy-at-pieces** [comment]
LGTM but seems there is a lot of duplicate code

**Review by Bishoy-at-pieces — COMMENTED** [review]
Can we run also bash not only shell?

**Comment by sambhavnoobcoder** [comment]
so far we are working on only shell scripts since Bash scripting is a subset of shell scripting . So most of the common bash commands would easily on the shell implementation itself , without any difference to the expected behaviour . Also , since shell is almost entirely compliant with the POSIX standards and bash is also nearly fully POSIX compliant ( bash version 4.0 and later have been made fully POSIX compliant) so we only have a very few scripts that might give error to the above implementation(Only a smaller niche of commands that may not have been upgraded , deprecations , etc .) . Hence I would suggest , that we could solve this with a simple try - catch statement , so that in case a command execution fails with the shell , it is then given to the bash implementation , and if that fails , it is concluded to be in an incorrect format . Considering the size of/ number of edge cases expected here Is small , I would think this to very feasible as well .

If you think this is a valid solution , I'll start coding it out , else if you have any other suggestion , I'd be glad to give that approach a look as well .

**Comment by sambhavnoobcoder** [comment]
these changes should address the required behaviour , kindly do check it out , and if everything is fine , I'll push the modified app.py as well .

File: src/pieces/commands/run_assets_command.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            asset = AssetsCommandsApi.get_asset_snapshot(uuid)
+            classification = cls.get_asset_classification(asset)
+            if classification in [AssetClassification.SHELL.value, AssetClassification.BASH.value]:
+                assets.append((f"{i}: {asset.name}", {"ITEM_INDEX": i, "UUID": uuid, "CLASSIFICATION": classification}))
+        
+        if not assets:
+            print("No shell or bash assets found")
+            return
+        
+        def custom_callback(**kwargs):
```
Comment (Bishoy-at-pieces) [inline src/pieces/commands/run_assets_command.py:33]: Can we rename it to something better than that?

**Review by Bishoy-at-pieces — APPROVED** [review]
LGTM

File: src/pieces/commands/run_assets_command.py (hunk 1 above)
Comment (sambhavnoobcoder) [reply src/pieces/commands/run_assets_command.py]: addressed issue in commit [c0d8d13](https://github.com/pieces-app/cli-agent/pull/169/commits/c0d8d139784b9a1296dbbb0771bb40013df3f785) . renamed custom_callback to open_and_execute_asset .  this change aims to make the function more self explanatory .

**Pull request merged**